src/main/java/com/microsoft/graph/httpcore/ChaosHttpHandler.java

```diff
@@ -0,0 +1,52 @@
+package com.microsoft.graph.httpcore;
+
+import java.io.IOException;
+import java.util.concurrent.ThreadLocalRandom;
+
+import com.microsoft.graph.httpcore.middlewareoption.MiddlewareType;
+import com.microsoft.graph.httpcore.middlewareoption.TelemetryOptions;
+
+import okhttp3.Interceptor;
+import okhttp3.MediaType;
+import okhttp3.Protocol;
+import okhttp3.Request;
+import okhttp3.Response;
+import okhttp3.ResponseBody;
+
+public class ChaosHttpHandler implements Interceptor {
```

Comment:
A quick summary of what this does could be useful, e.g. maybe a one-sentence summary + link to the full documentation of Interceptors

Reply:
added